File: java/singleton/GameSetting.java
public class GameSetting {
    private static GameSetting instance = null;

    private GameSetting() {
    }

    private int brightness = -1;
    private int width = -1;
    private int height = -1;
    private int fps = -1;

    public static GameSetting getInstance() {
        if (instance == null) {
            instance = new GameSetting();
        }
        return instance;
    }

    public int getBrightness() {
        return brightness;
    }

    public void setBrightness(int brightness) {
        this.brightness = brightness;
    }

    public int getWidth() {
        return width;
    }

    public void setWidth(int width) {
        this.width = width;
    }

    public int getHeight() {
        return height;
    }

    public void setHeight(int height) {
        this.height = height;
    }

    public int getFps() {
        return fps;
    }

    public void setFps(int fps) {
        this.fps = fps;
    }

    public void display() {
        System.out.println("Brightness: " + brightness);
        System.out.println("Width: " + width);
        System.out.println("Height: " + height);
        System.out.println("Fps: " + fps);
        System.out.println();
    }
}

File: cpp/singleton/GameSetting.cpp
#include "GameSetting.h"
GameSetting *GameSetting::instance = nullptr;
GameSetting::GameSetting()
{
    // ctor
}
GameSetting *GameSetting::getInstance()
{
    if (instance == nullptr)
    {
        instance = new GameSetting();
    }
    return instance;
}

int GameSetting::getBrightness()
{
    return brightness;
}

void GameSetting::setBrightness(int brightness)
{
    this->brightness = brightness;
}

int GameSetting::getVolume()
{
    return volume;
}

void GameSetting::setVolume(int volume)
{
    this->volume = volume;
}

int GameSetting::getWidth()
{
    return width;
}

void GameSetting::setWidth(int width)
{
    this->width = width;
}

int GameSetting::getHeight()
{
    return height;
}

void GameSetting::setHeight(int height)
{
    this->height = height;
}

int GameSetting::getFps()
{
    return fps;
}

void GameSetting::setFps(int fps)
{
    this->fps = fps;
}

void GameSetting::displaySetting()
{
    cout << "Brightness: " << this->brightness << endl;
    cout << "Volume: " << this->volume << endl;
    cout << "Width: " << this->width << endl;
    cout << "Height: " << this->height << endl;
    cout << "Fps: " << this->fps << endl;
    cout << endl;
}